MekHQ/src/mekhq/campaign/universe/eras/Eras.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (c) 2021 - The MegaMek Team. All Rights Reserved.
+ *
+ * This file is part of MekHQ.
+ *
+ * MekHQ is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * MekHQ is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MekHQ. If not, see <http://www.gnu.org/licenses/>.
+ */
+package mekhq.campaign.universe.eras;
+
+import megamek.common.annotations.Nullable;
+import megamek.common.util.fileUtils.MegaMekFile;
+import mekhq.MekHQ;
+import mekhq.MekHqConstants;
+import mekhq.MekHqXmlUtil;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.time.LocalDate;
+import java.util.Map;
+import java.util.SortedMap;
+import java.util.TreeMap;
+
+public class Eras {
+    //region Variable Declarations
+    private static Eras instance;
+
+    private SortedMap<LocalDate, Era> eras;
+    //endregion Variable Declarations
+
+    //region Constructors
+    private Eras() {
+        setEras(new TreeMap<>(LocalDate::compareTo));
+    }
+    //endregion Constructors
+
+    //region Getters/Setters
+    public static Eras getInstance() {
+        if (instance == null) {
+            setInstance(new Eras());
+        }
+
+        return instance;
+    }
+
+    public static void setInstance(final @Nullable Eras instance) {
+        Eras.instance = instance;
+    }
+
+    public SortedMap<LocalDate, Era> getEras() {
+        return eras;
+    }
+
+    private void setEras(final TreeMap<LocalDate, Era> eras) {
+        this.eras = eras;
+    }
+
+    public Era getEra(final LocalDate today) {
+        for (final Map.Entry<LocalDate, Era> entry : getEras().entrySet()) {
+            if (today.isBefore(entry.getKey())) {
```

Comment:
Why not make use of the TreeMap's functionality?

Reply:
What do you mean by that?

Reply:
TreeMap has a "floor" and a "ceiling" method, which takes a parameter of the key's data type and returns the nearest entry that's got either a lower or higher key.

Reply:
Because I didn't know about it. Works well, so made the changes. I also fixed the spellings of two eras.